warning: visibility attribute not supported in this configuration; ignored

--- origsrc/xproto-7.0.15/Xfuncproto.h.in	2007-07-07 07:54:49.000000000 -0500
+++ src/xproto-7.0.15/Xfuncproto.h.in	2009-06-29 15:49:42.463089200 -0500
@@ -85,7 +85,7 @@ in this Software without prior written a
 # define _X_ATTRIBUTE_PRINTF(x,y)
 #endif /* GNUC >= 4 */
 
-#if defined(__GNUC__) && (__GNUC__ >= 4)
+#if defined(__GNUC__) && (__GNUC__ >= 4) && !defined(__CYGWIN__)
 # define _X_EXPORT      __attribute__((visibility("default")))
 # define _X_HIDDEN      __attribute__((visibility("hidden")))
 # define _X_INTERNAL    __attribute__((visibility("internal")))
